### PR TITLE
Fix ListView selection mode changed

### PR DESCRIPTION
**Pull request by dr1rrb**
GitHub Issue: fixes https://github.com/unoplatform/uno/issues/2839

## Bugfix
Invalid visual state when changing a `ListView`'s selectuion mode while one or more items are selected.

## What is the current behavior?
When the `SelectionMode` is changed to `None` while at least on item is selected, the `ListView` tries to un-select them, but doing this it alter the collection that it is enumerating.

## What is the new behavior?
We create a clone of the selected items collection.

## PR Checklist
- [ ] ~~Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)~~
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI.Tests/Windows_UI_XAML_Controls/ListViewBaseTests/Given_ListViewBase.cs [hunk 1]
```diff
@@ -19,6 +19,9 @@ namespace Uno.UI.Tests.ListViewBaseTests
 	public class Given_ListViewBase
 	{
 #if !NETFX_CORE
+		// Make sure to have a valid custom theme set, so it won't try to read it from the Application.Current(<<== null).RequestedTheme
+		[TestInitialize] public void Init() => global::Uno.UI.ApplicationHelper.RequestedCustomTheme = "HighContrast";
```
Comment (jeromelaban) [inline src/Uno.UI.Tests/Windows_UI_XAML_Controls/ListViewBaseTests/Given_ListViewBase.cs]: We probably should set that at assembly level: https://github.com/unoplatform/uno/blob/d804b940617548f579c2826b7ba9e53c55eb554e/src/Uno.UI.Tests/Global.cs

Comment (davidjohnoliver) [reply src/Uno.UI.Tests/Windows_UI_XAML_Controls/ListViewBaseTests/Given_ListViewBase.cs]: Not sure it's a good idea to set this, some tests IIRC rely on the default theme being set. 

Where is it causing a problem if RequestedCustomTheme is not set?

Comment (dr1rrb) [reply src/Uno.UI.Tests/Windows_UI_XAML_Controls/ListViewBaseTests/Given_ListViewBase.cs]: It's in the generated code, while resolving the `ContentControlThemeFontFamily`:
```cs
public static FontFamily ContentControlThemeFontFamily
{
	get
	{
		// Custom themes defined for this resource: checking custom theme.
		var currentCustomTheme = global::Uno.UI.ApplicationHelper.RequestedCustomTheme;
		switch(currentCustomTheme)
		{
			case "HighContrast": return ContentControlThemeFontFamily___HighContrast;
		}

		// Element's RequestedTheme not supported yet. Fallback on Application's RequestedTheme.
		var currentTheme = global::Windows.UI.Xaml.Application.Current.RequestedTheme; <<<<==== Application.Current is null
		switch(currentTheme)
		{
			default: return ContentControlThemeFontFamily___Default;
			case global::Windows.UI.Xaml.ApplicationTheme.Light: return ContentControlThemeFontFamily___Light;
		}
	}
}
```

And I'm not convince it's good idea to put it at the assembly level, as it's only a patch to fix tests when run locally. I assume that on the CI the tests sequence make it work properly, so changing this value up-front might fail some other tests.

The real fix would be to generate a `null` proof code, and I actually started to to do that until I realized I was able to workaround the bug, without any risk of destabilization. Another solution would also be to have a non `null` `Application.Current`.

I've created another issue for this : https://github.com/unoplatform/uno/issues/2852

Comment (davidjohnoliver) [reply src/Uno.UI.Tests/Windows_UI_XAML_Controls/ListViewBaseTests/Given_ListViewBase.cs]: Aha I didn't read the comment properly, the real problem is that `Application.Current` is null. 

That's an easy fix:
```csharp
UnitTestsApp.App.EnsureApplication();
```

**Comment by jeromelaban** [comment]
@mergifyio backport release/beta/2.1

**Pull request merged**